x/wasm/simulation/proposals.go

```diff
@@ -239,20 +200,19 @@ func DefaultSimulateUpdateAdminProposalContractSelector(
 // Simulate update admin contract proposal
 //
 
-func SimulateUpdateAdminProposal(wasmKeeper WasmKeeper, contractSelector UpdateAdminContractSelector) simtypes.ContentSimulatorFn {
-	return func(r *rand.Rand, ctx sdk.Context, accs []simtypes.Account) simtypes.Content {
+func SimulateUpdateAdminProposal(wasmKeeper WasmKeeper, contractSelector UpdateAdminContractSelector) simtypes.MsgSimulatorFn {
+	return func(r *rand.Rand, ctx sdk.Context, accs []simtypes.Account) sdk.Msg {
 		simAccount, _ := simtypes.RandomAcc(r, accs)
 		ctAddress, _ := contractSelector(ctx, wasmKeeper, simAccount.Address.String())
 		if ctAddress == nil {
 			return nil
 		}
 
-		return types.NewUpdateAdminProposal(
-			simtypes.RandStringOfLength(r, 10),
-			simtypes.RandStringOfLength(r, 10),
-			simtypes.RandomAccounts(r, 1)[0].Address.String(),
-			ctAddress.String(),
-		)
+		return &types.MsgUpdateAdmin{
+			Sender:   "",
```

Comment:
Here and others: shouldn't this be set to the `authority` address? If we lease this empty, we can not compare it to the authority address set on the keeper for gov context. I am also a bit nervous about side effects and that this scenario could be created by a contract that ignores [authorization](https://github.com/CosmWasm/wasmd/blob/v0.31.0/x/wasm/keeper/handler_plugin.go#L89)

When we discussed empty address, I had CLI usage in mind only where the authority address would be fetched from chain.